numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -3066,6 +3066,32 @@ def add_file(self, path, kind):
         name = pathlib.Path(path).name
         self.add_data(data, kind, name)
 
+    def add_cu(self, cu, name):
+        """Add CUDA source in a string to the link. The name of the source
+        file should be specified in `name`."""
+        with driver.get_active_context() as ac:
+            dev = driver.get_device(ac.devnum)
+            cc = dev.compute_capability
+
+        program, log = nvrtc.compile(cu, name, cc, ltoir=self.lto)
+
+        if not self.lto and config.DUMP_ASSEMBLY:
+            # TODO: when linker is configured to generate LTOIR, assembly is not
+            # directly visible via the Linker API. We should provide `-ptx` as
+            # an additional flag to nvjitlink to generate PTX. However, this
+            # could break the linking pipeline. We need to investigate this further.
```

Comment:
To get PTX when doing LTO, [`get_cubin()`](https://github.com/NVIDIA/numba-cuda/blob/main/numba_cuda/numba/cuda/codegen.py#L174) will need to run the linker twice, once without `"-ptx"` (as it does now) and once again with `"-ptx"` (additionally) to get the PTX - it can then dump it in a similar way to how it's dumped here.